handlers/discount-api/src/discountEndpoint.ts

```diff
@@ -50,21 +80,49 @@ export const applyDiscountEndpoint = async (
 ) => {
 	const zuoraClient = await ZuoraClient.create(stage);
 
-	const { subscription, discount, dateToApply } = await getDiscountToApply(
-		stage,
-		headers,
-		subscriptionNumber,
-		zuoraClient,
-	);
-	return await applyDiscount(
+	const { subscription, discount, dateToApply, account } =
+		await getDiscountToApply(stage, headers, subscriptionNumber, zuoraClient);
+	console.log('Apply a discount to the subscription');
+	const discounted = await addDiscount(
 		zuoraClient,
 		subscriptionNumber,
-		subscription.termStartDate,
-		subscription.termEndDate,
-		dateToApply,
+		dayjs(subscription.termStartDate),
+		dayjs(subscription.termEndDate),
+		dayjs(dateToApply),
 		discount.productRatePlanId,
+	);
+
+	if (discounted.success) {
+		console.log('Discount applied successfully');
+	} else {
+		throw new Error('discount was not applied: ' + JSON.stringify(discounted));
+	}
```

Comment:
Totally subjective, but just to reduce the nesting (and hopefully increase readability) I'd probably put the throw in a condition and then have the console.log after:

```suggestion
	if (!discounted.success) {
	        throw new Error('discount was not applied: ' + JSON.stringify(discounted));
	}
	
	console.log('Discount applied successfully');
```